diff --git forkSrcPrefix/libavcodec/mlpdec.c forkDstPrefix/libavcodec/mlpdec.c
index e69e4442d25afcad51d1d7910fbb5798f824dc53..5f31fbab9073cfb2505ef9c8a35b959e75e67e70 100644
--- forkSrcPrefix/libavcodec/mlpdec.c
+++ forkDstPrefix/libavcodec/mlpdec.c
@@ -467,9 +467,9 @@ static int read_major_sync(MLPDecodeContext *m, GetBitContext *gb)
                 m->substream[2].mask = mh.channel_layout_thd_stream2;
             else
                 m->substream[2].mask = mh.channel_layout_thd_stream1;
-        if (mh.num_substreams == 2 && (!m->downmix_layout.nb_channels ||
+        if (mh.num_substreams <= 2 && (!m->downmix_layout.nb_channels ||
                                         m->downmix_layout.nb_channels > 2))
-            m->substream[1].mask = mh.channel_layout_thd_stream2;
+            m->substream[mh.num_substreams > 1].mask = mh.channel_layout_thd_stream2;
     }
 
     m->needs_reordering = mh.channel_arrangement >= 18 && mh.channel_arrangement <= 20;
